xline/src/storage/auth_store/store.rs

```diff
@@ -99,10 +101,39 @@ where
         }
     }
 
-    /// generate a next revision and save it into backend
-    fn commit_revision(&self) -> Result<(), ExecuteError> {
-        let _ignore = self.revision.next();
-        self.backend.save_revision(&self.revision)
+    /// generate a next revision and create a write operation to persistent it
+    fn op_inc_and_put_revision(&self) -> WriteOperation {
```

Comment:
I think we shouldn't expose some lower details, like `WriteOperation`,  to the upper layer. This goes against the law of Demeter. If we need to modify the storage engine layer for some reason in the future, this change will diffuse to the upper layer, like the Store layer and its upper layer.

Reply:
Fixed